### PR TITLE
Remove the hold for future months button

### PR DESCRIPTION
**Pull request by shall0pass**
The "Hold for future months" button appears to delay money that is in on-budget accounts by 12 months.  I personally don't see a use case for this function.  This PR removes the button.  If there is a use case I'm not aware of that depends on this functionality, please cancel this PR.

### CODE REVIEW

**Review by MatissJanis — COMMENTED** [review]
I actually use this functionality, but it's not critical to my workflow. So I'm not strictly against removing it.. :)

Do we have any client-side metrics for this functionality @rich-howell @jlongster ? Ideally we should make a decision based in the data. But if there is no data, then I'd err on the side of caution with removing features.

---

@shall0pass code-wise looks mostly OK. There are a couple more things that can be removed. For example: `holdForFutureMonths`. Would you mind reviewing your changes and seeing if there is any more dead weight we can delete?

File: packages/loot-core/src/server/budget/app.js [hunk 1]
```diff
@@ -12,15 +12,10 @@ app.method(
 );
 app.method('budget/set-zero', mutator(undoable(actions.setZero)));
 app.method('budget/set-3month-avg', mutator(undoable(actions.set3MonthAvg)));
-app.method('budget/set-all-future', mutator(undoable(actions.setAllFuture)));
```
Comment (MatissJanis) [inline packages/loot-core/src/server/budget/app.js:15]: This is still being used. You should either revert this removal or clean it up entirely.

Comment (shall0pass) [reply packages/loot-core/src/server/budget/app.js:15]: I'm looking.  I'm not particularly efficient at JS, so I'll be little slow.  Thanks for your patience.

Comment (MatissJanis) [reply packages/loot-core/src/server/budget/app.js:15]: No problem. Take all the time you need.

What I recommend is doing a global search for the function names you're removing and see what comes up. You can either do that in your IDE or in Github.

Here's an example search in Github: https://github.com/actualbudget/actual/search?q=set-all-future

By following this approach you'll find all the references to the functions and hence will be able to remove all of them.

Comment (shall0pass) [reply packages/loot-core/src/server/budget/app.js:15]: I saw this comment: "setAllFuture only applies to report budget type"
I'm not familiar with that Budget Type.  Are we sure this should be removed?

**Review by MatissJanis — APPROVED** [review]
I'm not 100% sure about the removal of `set-all-future` from a product perspective.

But from a code POV: looks good. :+1:

**Comment by shall0pass** [comment]
> I'm not 100% sure about the removal of `set-all-future` from a product perspective.
> 
> But from a code POV: looks good. 👍

Thanks for helping me through it.

**Review by jlongster — APPROVED** [review]
Looks great to me (sorry for the delay), thanks a lot for doing this!

**Comment by jlongster** [comment]
In the future, you don't need to generate the API bundle. That's only needed when we cut a release.

**Comment by jlongster** [comment]
Looks like there's a small formatting error; I can fix it tomorrow

**Pull request merged**